tools depending on rustc crates can't build with `rust.download-rustc=true` and `llvm.download-ci-llvm=false` options

When trying to build tools that rely on rustc crates using the `rust.download-rustc=true` and `llvm.download-ci-llvm=false` options, `ld` can not find the LLVM it needs.

```sh
$ x build miri
...
...
...
  = note: /usr/bin/ld: cannot find -lLLVM-18-rust-1.79.0-nightly: No such file or directory
          collect2: error: ld returned 1 exit status


error: could not compile `rustdoc-tool` (bin "rustdoc_tool_binary") due to 1 previous error
Build completed unsuccessfully in 0:01:02
```

I have also attempted to build `rustdoc` and `clippy` and they too failed due to the same issue. So I am quite sure this happens on any tool that uses rustc crates.

--- a/config.example.toml
+++ b/config.example.toml
@@ -50,7 +50,7 @@
 #
 # Note that many of the LLVM options are not currently supported for
 # downloading. Currently only the "assertions" option can be toggled.
-#download-ci-llvm = if rust.channel == "dev" { "if-unchanged" } else { false }
+#download-ci-llvm = if rust.channel == "dev" || rust.download-rustc != false { "if-unchanged" } else { false }
 
 # Indicates whether the LLVM build is a Release or Debug build
 #optimize = true

--- a/src/bootstrap/src/core/config/config.rs
+++ b/src/bootstrap/src/core/config/config.rs
@@ -2483,9 +2483,20 @@ impl Config {
                 llvm::is_ci_llvm_available(self, asserts)
             }
         };
+
         match download_ci_llvm {
-            None => self.channel == "dev" && if_unchanged(),
-            Some(StringOrBool::Bool(b)) => b,
+            None => {
+                (self.channel == "dev" || self.download_rustc_commit.is_some()) && if_unchanged()
+            }
+            Some(StringOrBool::Bool(b)) => {
+                if !b && self.download_rustc_commit.is_some() {
+                    panic!(
+                        "`llvm.download-ci-llvm` cannot be set to `false` if `rust.download-rustc` is set to `true` or `if-unchanged`."
+                    );
+                }
+
+                b
+            }
             // FIXME: "if-available" is deprecated. Remove this block later (around mid 2024)
             // to not break builds between the recent-to-old checkouts.
             Some(StringOrBool::String(s)) if s == "if-available" => {

